daffodil-core/src/main/scala/org/apache/daffodil/dsom/RestrictionUnion.scala

```diff
@@ -166,16 +175,126 @@ final class Restriction(xmlArg: Node, val simpleType: SimpleTypeDefBase)
+
+  lazy val facetValueSet: RepValueSet[AnyRef] = {
+    val initAns: (RangeBound[BigInt], RangeBound[BigInt]) = (new RangeBound(Maybe.Nope, false), new RangeBound(Maybe.Nope, false))
+    val range = combinedBaseFacets.foldLeft(initAns)({
+      case (acc, (facetType, facetValue)) =>
+        lazy val valueAsBigInt = BigInt(facetValue)
+        val (maybeBound, isMax, isInclusive) = facetType match {
+          case Facet.maxExclusive => (Maybe(valueAsBigInt), true, false)
+          case Facet.maxInclusive => (Maybe(valueAsBigInt), true, true)
+          case Facet.minExclusive => (Maybe(valueAsBigInt), false, false)
+          case Facet.minInclusive => (Maybe(valueAsBigInt), false, true)
+          case _                  => (Maybe.Nope, false, false)
+        }
+        if (maybeBound.isEmpty) {
+          acc
+        } else {
+          if (isMax) {
+            val oldMax = acc._2
+            if (!oldMax.isEmpty) {
+              SDE("Cannot define multiple max bounds on restriction")
+            }
+            val newMax = new RangeBound(maybeBound, isInclusive)
+            (acc._1, newMax)
+          } else {
+            val oldMin = acc._1
+            if (!oldMin.isEmpty) {
+              SDE("Cannot define multiple min bounds on restriction")
+            }
+            val newMin = new RangeBound(maybeBound, isInclusive)
+            (newMin, acc._2)
+          }
+        }
+    })
+    RepValueSetCompiler.compile(Seq(), Seq(range.asInstanceOf[(RangeBound[AnyRef], RangeBound[AnyRef])]))
+  }
+  lazy val repValueSet: RepValueSet[AnyRef] = {
+    val subsets = enumerations.map(_.optRepValueSet).filter(_.isDefined).map(_.get)
+    if (subsets.length != 0 && subsets.length != enumerations.length) {
+      context.SDE("If one enumeration value defines a repValue, then all must define a repValue")
+    }
+    val fromEnums = subsets.fold(RepValueSetCompiler.empty)((a, b) => a.merge(b))
+    if (enumerations.length > 0) {
+      fromEnums
+    } else {
+      //We are some form of identity mapping, so our facetValues are also our repValues
+      facetValueSet
+    }
+  }
+  lazy val optRepValueSet = if (repValueSet.isEmpty) None else Some(repValueSet)
+
+  lazy val logicalValueSet: RepValueSet[AnyRef] = {
+    val subsets = enumerations.map(_.logicalValueSet)
+    if (subsets.length != 0 && subsets.length != enumerations.length) {
+      context.SDE("If one enumeration value defines a repValue, then all must define a repValue")
+    }
+    val fromEnums = subsets.fold(RepValueSetCompiler.empty)((a, b) => a.merge(b))
+    if (enumerations.length > 0) {
+      fromEnums
+    } else {
+      //We are some form of identity mapping, so our facetValues are also our repValues
+      facetValueSet
+    }
+  }
+  lazy val optLogicalValueSet:Option[RepValueSet[AnyRef]] = if(logicalValueSet.isEmpty) None else Some(logicalValueSet)
+
+  lazy val optRepType:Option[SimpleTypeBase] = {
+    enumerations.map(_.optRepTypeFactory)
+    ???
```

Comment:
This should be addressed before being merged.

Reply:
Not sure what the original intent of this field was, so I'm just going to remove it.